token-account-based/chaincode-go/chaincode/token_contract.go

```diff
@@ -172,3 +194,26 @@ func (s *SmartContract) ClientAccountID(ctx contractapi.TransactionContextInterf
 
 	return clientAccountID, nil
 }
+
+// TotalSupply returns the total token supply
+func (s *SmartContract) TotalSupply(ctx contractapi.TransactionContextInterface) (int, error) {
+
+	// Retrieve total supply of tokens from state of smart contract
+	totalSupplyBytes, err := ctx.GetStub().GetState(totalSupplyKey)
+	if err != nil {
+		return 0, fmt.Errorf("failed to retrieve total token supply: %v", err)
+	}
+
+	var totalSupply int
+
+	// If no tokens have been minted, return 0
+	if totalSupplyBytes == nil {
+		totalSupply = 0
+	} else {
+		totalSupply, _ = strconv.Atoi(string(totalSupplyBytes)) // Error handling not needed since Itoa() was used when setting the account balance, guaranteeing it was an integer.
```

Comment:
```suggestion
		totalSupply, _ = strconv.Atoi(string(totalSupplyBytes)) // Error handling not needed since Itoa() was used when setting the totalSupply, guaranteeing it was an integer.
```